src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -18629,5 +18629,510 @@ public static void Main()
+                // (2,15): error CS1057: 'R.R(R)': static classes cannot contain protected members
+                // static record R(int I)
+                Diagnostic(ErrorCode.ERR_ProtectedInStatic, "R").WithArguments("R.R(R)").WithLocation(2, 15),
+                // (2,15): error CS1057: 'R.EqualityContract': static classes cannot contain protected members
+                // static record R(int I)
+                Diagnostic(ErrorCode.ERR_ProtectedInStatic, "R").WithArguments("R.EqualityContract").WithLocation(2, 15),
+                // (2,15): error CS1057: 'R.EqualityContract.get': static classes cannot contain protected members
+                // static record R(int I)
+                Diagnostic(ErrorCode.ERR_ProtectedInStatic, "R").WithArguments("R.EqualityContract.get").WithLocation(2, 15),
+                // (2,21): error CS0708: 'R.I': cannot declare instance members in a static type
+                // static record R(int I)
+                Diagnostic(ErrorCode.ERR_InstanceMemberInStaticClass, "I").WithArguments("R.I").WithLocation(2, 21),
+                // (4,5): error CS0710: Static types cannot have instance constructors
+                //     R() : this(0) { }
+                Diagnostic(ErrorCode.ERR_ConstructorInStaticClass, "R").WithLocation(4, 5),
+                // (5,6): error CS0711: Static types cannot contain destructors
+                //     ~R() { }
+                Diagnostic(ErrorCode.ERR_DestructorInStaticClass, "R").WithArguments("R.~R()").WithLocation(5, 6)
+                );
+        }
+
+        [Fact]
+        public void RecordWithPartialMethodExplicitImplementation()
+        {
+            var source =
+@"record R
+{
+    partial void M();
+}";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (3,18): error CS0751: A partial method must be declared within a partial type
+                //     partial void M();
+                Diagnostic(ErrorCode.ERR_PartialMethodOnlyInPartialClass, "M").WithLocation(3, 18)
+                );
+        }
+
+        [Fact]
+        public void RecordWithMultipleBaseTypes()
+        {
+            var source = @"
+record Base1;
+record Base2;
+record R : Base1, Base2
+{
+}";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (4,19): error CS1721: Class 'R' cannot have multiple base classes: 'Base1' and 'Base2'
+                // record R : Base1, Base2
+                Diagnostic(ErrorCode.ERR_NoMultipleInheritance, "Base2").WithArguments("R", "Base1", "Base2").WithLocation(4, 19)
+                );
+        }
+
+        [Fact]
+        public void RecordWithInterfaceBeforeBase()
+        {
+            var source = @"
+record Base;
+interface I { }
+record R : I, Base;
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (4,15): error CS1722: Base type 'Base' must come before any interfaces
```

Comment:
Since "base type" [includes](https://github.com/dotnet/csharplang/blob/master/spec/classes.md#type-parameter-constraints) interfaces, I'll revert to "base class" here as well.